### DISCUSSION TITLE
How to deInstall lonhorn from OpenShift - delete namespace longhorn-system  hang

### DISCUSSION THREAD

**Discussion by n00bsi** [longhorn/longhorn#8712] [General]
Hi,

try do deinstall lonhorn

```
helm uninstall longhorn -n longhorn-system
Error: 1 error occurred:
        * job failed: BackoffLimitExceeded

```

then I tryed:

```

oc delete ns longhorn-system
namespace "longhorn-system" deleted

```

but this hang and did not come to end... 

what did I wrong ?

```
oc -n longhorn-system get svc
No resources found in longhorn-system namespace.

```

but 

```
oc get ns | grep long
longhorn-system                                    Terminating   14h

```

**Comment by c3y1huang** [comment]
Check out https://longhorn.io/docs/1.6.2/deploy/uninstall/#troubleshooting.

**Comment by n00bsi** [comment]
thanks.... found a soltuion here:

https://www.ibm.com/support/pages/cloud-pak-security-how-delete-project-stuck-terminating-status

```
oc get namespace <Project> -o json > longhorn-ns.json
vi longhorn-ns.json

```
remove the line: with "kubernetes"

from:

```
       ...
    },
    "spec": {
        "finalizers": [
            "kubernetes"
        ]
    },
   ....

```

to:

```

      ...
    },
    "spec": {
        "finalizers": []
    },
        ...
```

Login to console -> click on username -> copy login command
get the token

`curl -k -H "Content-Type: application/json" -H "Authorization: Bearer ADD_TOKEN_HERE" -X PUT --data-binary @longhorn-ns.json  https://<OPENSHIFT_CONSOLE_ADDRESS>:6443/api/v1/namespaces/longhorn-system/finalize
`
